PublicGitArchive/pga/cmd/cache.go

```diff
@@ -16,46 +15,68 @@ const (
 	indexName = "latest.csv.gz"
 )
 
-// copy checks whether a new version of the file in url exists and downloads it
-// to dest. It returns true when the file exists, and an error when it was not possible
-// to update it.
-func copy(dest, source FileSystem, name string) (bool, error) {
+// updateCache checks whether a new version of the file in url exists and downloads it
+// to dest. It returns an error when it was not possible to update it.
+func updateCache(dest, source FileSystem, name string) error {
```

Comment:
Is naming like `copyMaybe` a thing in Golang? Might come handy explaining this case

Reply:
`copyMaybe` doesn't sound idiomatic, I've never seen it before.
I think `updateCache` is pretty clear, but I'm happy to modify it if you think it's not.

Reply:
Up to you! It's just a common naming convention in many programming languages that origins from functional programming, to signify a function that have conditional side-effects.

See examples in: 
  - `maybe_download_iris_data` in [Tensorflow](https://github.com/tensorflow/tensorflow/blob/d4ad9c73969c45d1a224ebfc43eb645b9860216b/tensorflow/examples/learn/iris.py#L38) Python
 - `maybe_download_dbpedia` in [Tensorflow](https://github.com/tensorflow/tensorflow/blob/7f53659bc67bba5567ea3f0b69710329843e0228/tensorflow/contrib/learn/python/learn/datasets/produce_small_datasets.py#L32) Python
 - `resizeMeMaybe` in [Guava](https://github.com/google/guava/blob/7abf5990b29f19a7e8c458b63af73c88523e1820/android/guava/src/com/google/common/collect/ObjectCountHashMap.java#L326) Java

Quick search over stdlib shows some hits in both public and private APIs:
 - `maybeOpenNewConnections ` https://github.com/golang/go/blob/master/src/database/sql/sql.go#L980
 - `GetMaybeInsecure` https://github.com/golang/go/blob/178307c3a72a9da3d731fecf354630761d6b246c/src/cmd/go/internal/web/http.go#L76